hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/x509/certificate/client/TestRootCaRotationPoller.java

```diff
@@ -77,24 +77,18 @@ public void testPollerDoesNotInvokeRootCaProcessor() throws Exception {
     certsFromScm.add(CertificateCodec.getPEMEncodedString(knownCert));
     RootCaRotationPoller poller = new RootCaRotationPoller(secConf,
         knownCerts, scmSecurityClient, "");
-
+    //When the scm returns the same set of root ca certificates, and they poll
+    //for them
     Mockito.when(scmSecurityClient.getAllRootCaCertificates())
         .thenReturn(certsFromScm);
-    AtomicBoolean atomicBoolean = new AtomicBoolean();
-    atomicBoolean.set(false);
-    poller.addRootCARotationProcessor(
-        certificates -> CompletableFuture.supplyAsync(() -> {
-          atomicBoolean.set(true);
-          Assertions.assertEquals(certificates.size(), 2);
-          return null;
-        }));
-    poller.run();
-    Assertions.assertThrows(TimeoutException.class, () ->
-        GenericTestUtils.waitFor(atomicBoolean::get, 50, 5000));
+    //Then assume that the rootCaRotationProcessors weren't invoked
+    Assertions.assertFalse(poller.pollRootCas());
```

Comment:
If we just check the poller.pollRootCas() status, then this test doesn't match the test function name, as the invoke is not checked.  The way used in testPollerRetriesAfterFailure() can be used here too.